src/encoded/static/components/search.js

```diff
@@ -277,39 +277,64 @@ globals.listingViews.register(Biosample, 'Biosample');
     const { context: result, cartControls, mode } = props;
     let synchronizations;
 
-    // Determine whether object is Experiment or FunctionalCharacterizationExperiment.
+    // Determine whether object is Experiment, FunctionalCharacterizationExperiment, or TransgenicEnhancerExperiment.
     const experimentType = result['@type'][0];
     const isFunctionalExperiment = experimentType === 'FunctionalCharacterizationExperiment';
-    const displayType = isFunctionalExperiment ? 'Functional Characterization Experiment' : 'Experiment';
+    const isEnhancerExperiment = experimentType === 'TransgenicEnhancerExperiment';
+    let displayType;
+    if (isFunctionalExperiment) {
+        displayType = 'Functional Characterization Experiment';
+    } else if (isEnhancerExperiment) {
+        displayType = 'Transgenic Enhancer Experiment';
+    } else {
+        displayType = 'Experiment';
+    }
 
     // Collect all biosamples associated with the experiment. This array can contain duplicate
     // biosamples, but no null entries.
     let biosamples = [];
     const treatments = [];
 
-    if (result.replicates && result.replicates.length > 0) {
-        biosamples = _.compact(result.replicates.map(replicate => replicate.library && replicate.library.biosample));
-        // flatten treatment array of arrays
-        _.compact(biosamples.map(biosample => biosample.treatments)).forEach(treatment => treatment.forEach(t => treatments.push(t)));
+    if (isEnhancerExperiment) {
+        if (result.biosamples && result.biosamples.length > 0) {
+            biosamples = _.compact(result.biosamples.map(sample => biosamples));
+        }
+    } else {
+        if (result.replicates && result.replicates.length > 0) {
+            biosamples = _.compact(result.replicates.map(replicate => replicate.library && replicate.library.biosample));
+        }
+    // flatten treatment array of arrays
+    _.compact(biosamples.map(biosample => biosample.treatments)).forEach(treatment => treatment.forEach(t => treatments.push(t)));
     }
 
     // Get all biosample organism names
     const organismNames = biosamples.length > 0 ? BiosampleOrganismNames(biosamples) : [];
 
     // Collect synchronizations
-    if (result.replicates && result.replicates.length > 0) {
-        synchronizations = _.uniq(result.replicates.filter(replicate =>
-            replicate.library && replicate.library.biosample && replicate.library.biosample.synchronization
-        ).map((replicate) => {
-            const biosample = replicate.library.biosample;
-            return `${biosample.synchronization}${biosample.post_synchronization_time ? ` + ${biosample.age_display}` : ''}`;
-        }));
+    if (isEnhancerExperiment) {
+        if (result.biosamples && result.biosamples.length > 0) {
+            synchronizations = _.uniq(result.biosamples.filter(sample =>
+            sample && sample.synchronization
+            ).map((sample) => {
+                const biosample = sample;
```

Comment:
You can just use `sample` in the return line below to avoid this assignment, or rename the `sample` parameter to `biosample`. We often just use the singular form of the plural array name as the parameter to things like `map`.